server/src/main/java/io/crate/metadata/Reference.java

```diff
@@ -46,7 +46,7 @@ public interface Reference extends Symbol {
 
     RowGranularity granularity();
 
-    int position();
+    Integer position();
```

Comment:
Shouldn't the position keep being non-nullable?

If we ensure the columns have a position in the mapping, the references we create from it should always have one?

Reply:
We could probably use a -1 in the `AlterTableAddColumnAnalyzer.resolveField`  logic to avoid the streaming change (which becomes more complicated with the backports)